app/assets/javascripts/pageflow/before_after/widget.js

```diff
@@ -28,146 +28,161 @@
+            $(obj).css({'top':'0','left':leftOffset+'px'});
+          }
+        };
 
-        var beforeImg = $('.before_image', obj);
-        var afterImg = $('.after_image', obj);
+        imgWidth = beforeImg.attr("data-width");
+        imgHeight = beforeImg.attr("data-height");
+        var imgRatio = imgWidth / imgHeight;
 
-        var section = $(obj).parentsUntil(".backgroundArea").last();
 
-        var imgWidth;
-        var imgHeight;
+        $(obj).parent().toggleClass('image-is-landscape', imgRatio > 1);
+        $(obj).parent().toggleClass('image-is-portrait', imgRatio <= 1);
 
-        var refresh = function() {
-          var resizeAndPositionContainer = function() {
-            var screenRatio = section.width() / section.height();
+        var containerHeight = imgHeight / imgWidth * 100;
+        $(obj)
+          .css({'overflow':'visible','position':'relative','padding':'0','width':'100%', 'padding-top': containerHeight+"%"});
 
-            if(screenRatio < imgRatio) {
-              $(obj).parent().css({'width':'100%','height':'auto'});
-              var topOffset = (section.height() - $(obj).parent().height()) / 2;
-              $(obj).css({'top':topOffset+'px','left':'0'});
-            }
-            else {
-              $(obj).parent().css({'width': (section.height() * imgRatio)  + 'px','height':'100%'});
-              var leftOffset = (section.width() - $(obj).parent().width()) / 2;
-              $(obj).css({'top':'0','left':leftOffset+'px'});
-            }
-          };
+        resizeAndPositionContainer();
 
-          imgWidth = beforeImg.attr("data-width");
-          imgHeight = beforeImg.attr("data-height");
-          var imgRatio = imgWidth / imgHeight;
+        bb = obj.get()[0].getBoundingClientRect();
+        barOffset =  0;
+        dragwrapper.draggable();
+        dragwrapper.draggable('option', {axis: 'x', containment:[bb.left-barOffset, bb.top, bb.right-barOffset, bb.bottom],drag:drag,stop:drag });
+      };
 
+      this.refresh = refresh;
 
-          $(obj).parent().toggleClass('image-is-landscape', imgRatio > 1);
-          $(obj).parent().toggleClass('image-is-portrait', imgRatio <= 1);
+      beforeImg.attr('id','beforeimage'+randID);
+      afterImg.attr('id','afterimage'+randID);
 
-          var containerHeight = imgHeight / imgWidth * 100;
-          $(obj)
-            .css({'overflow':'visible','position':'relative','padding':'0','width':'100%', 'padding-top': containerHeight+"%"});
+      var bb;
+      var barOffset;
+      var beforeWrapper = $('.before_wrapper', obj);
 
-          resizeAndPositionContainer();
+      // Create an inner div wrapper (dragwrapper) to hold the images.
+      var dragwrapper = document.createElement('div');
+      dragwrapper.id = 'dragwrapper' + randID;
+      dragwrapper.className = 'dragwrapper';
+      dragwrapper.title = I18n.t('pageflow.public.before_after.dragwrapper.title', {
+        locale: pageflow.entry.configuration.get('locale')
```

Comment:
You are absolutely right! I neglected to do so for the later changes in my code.

I will make sure to do at least a browser check, and try to setup Travis as well if there are headless tests missing. Since I know myself, and I am pretty lazy. Apologies for this one!